#Interesting papers, software and new developments

### Trim Galore

http://www.bioinformatics.babraham.ac.uk/projects/trim_galore/

>A wrapper tool around Cutadapt and FastQC to consistently apply quality and adapter trimming to FastQ files, with some extra functionality for MspI-digested RRBS-type (Reduced Representation Bisufite-Seq) libraries.

### Trim Galore

http://www.bioinformatics.babraham.ac.uk/projects/trim_galore/

>A wrapper tool around Cutadapt and FastQC to consistently apply quality and adapter trimming to FastQ files, with some extra functionality for MspI-digested RRBS-type (Reduced Representation Bisufite-Seq) libraries.

### EPGA2

https://bioinformatics.oxfordjournals.org/content/31/24/3988.full

https://github.com/bioinfomaticsCSU/EPGA2

Relatively new de novo assembly algo. 


### TransRate

http://hibberdlab.com/transrate/

https://github.com/Blahah/transrate

>Transrate is software for de-novo transcriptome assembly quality analysis. It examines your assembly in detail and compares it to experimental evidence such as the sequencing reads, reporting quality scores for contigs and assemblies. This allows you to choose between assemblers and parameters, filter out the bad contigs from an assembly, and help decide when to stop trying to improve the assembly.


### KmerStream

http://bioinformatics.oxfordjournals.org/content/30/24/3541.full

https://github.com/pmelsted/KmerStream


Simple estimatation of sequencing errors + genome size, using only the aggregate statistics reported by KmerStream and validate the 
accuracy on sequences from a PhiX control

### qaTools

https://github.com/CosteaPaul/qaTools

>1. qaCompute
>   Computes normal and span coverage from a bam/sam file.
>   Also counts unmapped and sub-par quality reads.
>2. removeUnmapped
>   Remove unmapped and sub-par quality reads from a bam/sam file.
>3. computeInsertSizeHistogram
>   Compute the insert size distribution from a bam/sam file.

### SuRankCo

http://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0644-7

>We present SuRankCo, which relies on a machine learning approach to predict quality scores for contigs and to enable the ranking of contigs within an assembly. The result is a sorted contig set which allows selective contig usage in downstream analysis. Benchmarking on datasets with known ground truth shows promising sensitivity and specificity and favorable comparison to existing methodology.

<<<<<<< HEAD
### qaTools

https://github.com/CosteaPaul/qaTools

>1. qaCompute
>   Computes normal and span coverage from a bam/sam file.
>   Also counts unmapped and sub-par quality reads.
>2. removeUnmapped
>   Remove unmapped and sub-par quality reads from a bam/sam file.
>3. computeInsertSizeHistogram
>   Compute the insert size distribution from a bam/sam file.

### SuRankCo

http://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-015-0644-7

>We present SuRankCo, which relies on a machine learning approach to predict quality scores for contigs and to enable the ranking of contigs within an assembly. The result is a sorted contig set which allows selective contig usage in downstream analysis. Benchmarking on datasets with known ground truth shows promising sensitivity and specificity and favorable comparison to existing methodology.

=======
>>>>>>> master
### iWGS

https://github.com/zhouxiaofan1983/iWGS/

>iWGS is an automated pipeline for guiding the choice of appropriate sequencing strategy and assembly protocols. iWGS seamlessly integrates the four key steps of a de novo genome sequencing project: data generation (through simulation), data quality control, de novo assembly, and assembly evaluation and validation. The last three steps can also be applied to the analysis of real data. iWGS is designed to enable the user to have great flexibility in testing the range of experimental designs available for genome sequencing projects, and supports all major sequencing technologies and popular assembly tools.
<<<<<<< HEAD

=======
>>>>>>> master

### Recycler

https://github.com/rozovr/Recycler

Using dBJ models (eg SPAdes scaffolds files) to find plasmids and other small-circular genomic elements


### Bandage

http://bioinformatics.oxfordjournals.org/content/31/20/3350

http://rrwick.github.io/Bandage/

>Bandage is a program for visualising de novo assembly graphs. By displaying connections which are not present in the contigs file, Bandage opens up new possibilities for analysing de novo assemblies


### pileup.js

https://github.com/hammerlab/pileup.js

Web-based graphical view of sequence reads and SNVs


### GetDistr

https://github.com/ksahlin/GetDistr

>Reads from paired-end and mate-pair libraries are often utilized to find structural variation in genomes, and one common approach is to use their fragment length for detection. After aligning read-pairs to the reference, read-pair distances are analyzed for statistically significant deviations. However, previously proposed methods are based on a simplified model of observed fragment lengths that does not agree with data. We show how this model limits statistical analysis of identifying variants and propose a new model, by adapting a model we have previously introduced for contig scaffolding, which agrees with data.

###Neptune

http://biorxiv.org/content/early/2016/01/08/032227

Microbial Signature detection (analagous to MLST) at genomic-scale

### Hipmer

http://gauss.cs.ucsb.edu/~aydin/sc15_genome.pdf

### MyPro : pipeline for prokaryotic genome assembly

http://www.sciencedirect.com/science/article/pii/S0167701215001207

###  https://www.jstage.jst.go.jp/article/imt/10/1/10_175/_pdf

### A comparison among different assemblers

 http://link.springer.com/chapter/10.1007/978-3-662-47901-8_4
