#Interesting papers, software and new developments


### EPGA2

https://bioinformatics.oxfordjournals.org/content/31/24/3988.full

https://github.com/bioinfomaticsCSU/EPGA2

Relatively new de novo assembly algo. 


### TransRate

http://hibberdlab.com/transrate/

https://github.com/Blahah/transrate

>Transrate is software for de-novo transcriptome assembly quality analysis. It examines your assembly in detail and compares it to experimental evidence such as the sequencing reads, reporting quality scores for contigs and assemblies. This allows you to choose between assemblers and parameters, filter out the bad contigs from an assembly, and help decide when to stop trying to improve the assembly.


### KmerStream

http://bioinformatics.oxfordjournals.org/content/30/24/3541.full

https://github.com/pmelsted/KmerStream


Simple estimatation of sequencing errors + genome size, using only the aggregate statistics reported by KmerStream and validate the accuracy on sequences from a PhiX control

### qaTools

https://github.com/CosteaPaul/qaTools

>1. qaCompute
>   Computes normal and span coverage from a bam/sam file.
>   Also counts unmapped and sub-par quality reads.
>2. removeUnmapped
>   Remove unmapped and sub-par quality reads from a bam/sam file.
>3. computeInsertSizeHistogram
>   Compute the insert size distribution from a bam/sam file.


### Recycler

https://github.com/rozovr/Recycler

Using dBJ models (eg SPAdes scaffolds files) to find plasmids and other small-circular genomic elements


### Bandage

http://bioinformatics.oxfordjournals.org/content/31/20/3350

http://rrwick.github.io/Bandage/

>Bandage is a program for visualising de novo assembly graphs. By displaying connections which are not present in the contigs file, Bandage opens up new possibilities for analysing de novo assemblies


### pileup.js

https://github.com/hammerlab/pileup.js

Web-based graphical view of sequence reads and SNVs


### GetDistr

https://github.com/ksahlin/GetDistr

>Reads from paired-end and mate-pair libraries are often utilized to find structural variation in genomes, and one common approach is to use their fragment length for detection. After aligning read-pairs to the reference, read-pair distances are analyzed for statistically significant deviations. However, previously proposed methods are based on a simplified model of observed fragment lengths that does not agree with data. We show how this model limits statistical analysis of identifying variants and propose a new model, by adapting a model we have previously introduced for contig scaffolding, which agrees with data.

###Neptune

http://biorxiv.org/content/early/2016/01/08/032227

Microbial Signature detection (analagous to MLST) at genomic-scale

